src/Components/ChampionInfoComp/ChampionInfoComp.js

```diff
@@ -13,7 +13,7 @@ class ChampionInfoComp extends React.Component {
         if (this.props.champion) {  // champion was chosen
             if (!this.state.loadedPost || (this.state.loadedPost && this.state.loadedPost.title !== this.props.champion)) { // prevent infinite loops
                 let spell_types = ["Q", "W", "E", "R"];
-                let url = `http://ddragon.leagueoflegends.com/cdn/6.24.1/data/en_US/champion/${this.props.champion}.json`;
+                let url = `http://ddragon.leagueoflegends.com/cdn/9.14.1/data/en_US/champion/${this.props.champion}.json`;
                 axios.get(url).then( res => { // get one champion info
```

Comment:
Let's try to extract every single data fetching to a different module named: APIGateway.js, which will handle every server-related fetch for us.

Reply:
Also, use async-await https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function